apps/desktop-wallet/src/api/apiDataHooks/useAddressesTokensWorth.ts

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2018 - 2024 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { AddressHash, calculateAmountWorth, TOKENS_QUERY_LIMIT } from '@alephium/shared'
+import { useQueries } from '@tanstack/react-query'
+import { chunk } from 'lodash'
+import { useMemo } from 'react'
+
+import useAddressesPricedFTs from '@/api/apiDataHooks/useAddressesPricedFTs'
+import useAddressesTokensBalancesTotal from '@/api/apiDataHooks/useAddressesTokensBalancesTotal'
+import { flatMapCombine } from '@/api/apiDataHooks/utils'
+import { tokensPriceQuery } from '@/api/queries/priceQueries'
+import { useAppSelector } from '@/hooks/redux'
+import { TokenId } from '@/types/tokens'
+
+interface AddressesTokensWorth {
+  data: Record<TokenId, number | undefined>
+  isLoading: boolean
+}
+
+const useAddressesTokensWorth = (addressHash?: AddressHash): AddressesTokensWorth => {
+  const { data: tokenPrices, isLoading: isLoadingTokenPrices } = useAddressesTokensPrices()
+  const { data: tokensWithPrice, isLoading: isLoadingTokensWithPrice } = useAddressesPricedFTs(addressHash)
```

Comment:
Just a random thought as I'm reading this, I wonder if it wouldn't be clearer and in the end more efficient (?) to get rid of the very specific `useAddressesPricedFTs` and instead have a hook that returns the the addresses tokens with their price (when available) 🤔 
It's linked to a previous discussion we had, about maybe having a simple to use `useAddressesTokens` which is a dictionary of tokens grouped by type...

Reply:
+ a variant hook which return a flat array with the tokens.

Reply:
Discussed on Slack